### PR TITLE
LPS-181448: Add integration test in ObjectEntryRelatedObjectsResourceTest to add nested System Object entries with Custom Fields and fix ObjectRelationshipExtensionProvider nested entities management

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Invoke to `setExtendedProperties` method only with the extended properties expected by each ExtensionProvider. This will be very useful for future management with multiple depth nested entities.
- Add test cases for creating and updating Custom Object Entries and their related Syste Object Entries with a Custom Object Field.
- Remove two Ignore tags.

**How to reproduce?**

- Deploy `object-api` and `object-service`
- Deploy `object-rest-api` and `object-rest-impl`

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-181448

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java [hunk 1]
```diff
@@ -146,21 +148,31 @@ protected void setExtendedProperties(
...
-				_log.error(exception);
+
+			if (MapUtil.isNotEmpty(extendedProperties)) {
+				try {
+					extensionProvider.setExtendedProperties(
+						user.getCompanyId(), user.getUserId(), className,
+						entity, extendedProperties);
+				}
+				catch (Exception exception) {
+					_log.error(exception);
```
Comment (ccorreagg) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java:174]: Not sure if we should log and ignore the error or throws the exception instead.

Why is that code there? What kind of errors could happen?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 2]
```diff
@@ -543,6 +595,7 @@ public void testPutCustomObjectEntryWithNestedSystemObjectEntry()
 		// Many to many
 
 		_testPutCustomObjectEntryWithNestedSystemObjectEntry(
+			null, null,
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:598]: I think it makes no sense having more tests, as the one with the custom field should cover the case without custom field, right?

**Review by ccorreagg — COMMENTED** [review]
Some questions

File: modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java:174]: As it was explained by Objects team, this code could be removed

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 3]
```diff
@@ -511,9 +518,21 @@ public void testPostCustomObjectEntryWithInvalidNestedSystemObjectEntries()
 	public void testPostCustomObjectEntryWithNestedSystemObjectEntry()
 		throws Exception {
 
+		_objectFieldLocalService.addCustomObjectField(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:521]: Could we move this functionality to one of the TestUtils we are using?

I think it is going to be useful for testing system objects

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 4]
```diff
@@ -511,9 +518,21 @@ public void testPostCustomObjectEntryWithInvalidNestedSystemObjectEntries()
...
+			null, TestPropsValues.getUserId(), 0,
+			_userSystemObjectDefinition.getObjectDefinitionId(),
+			ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+			ObjectFieldConstants.DB_TYPE_STRING, false, true, "",
+			LocalizedMapUtil.getLocalizedMap(_CUSTOM_OBJECT_FIELD_NAME_1),
+			false, _CUSTOM_OBJECT_FIELD_NAME_1, false, false,
+			_getObjectFieldSettings("Text"));
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:528]: Please, pass the business type constant instead

File: modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java [hunk 5]
```diff
@@ -146,22 +146,27 @@ protected void setExtendedProperties(
...
+						propertyKey, (Serializable)entry.getValue());
+				}
+			}
+
+			if (MapUtil.isNotEmpty(extendedProperties)) {
 				extensionProvider.setExtendedProperties(
 					user.getCompanyId(), user.getUserId(), className, entity,
 					extendedProperties);
 			}
-			catch (Exception exception) {
```
[the remark this answers is not in the document]
Comment (ccorreagg) [reply modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java:162]: (For the following PRs, not this one)

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java [hunk 6]
```diff
@@ -189,14 +189,10 @@ public void setExtendedProperties(
 
 			ObjectRelationship objectRelationship =
 				_objectRelationshipLocalService.
-					fetchObjectRelationshipByObjectDefinitionId1(
+					getObjectRelationshipByObjectDefinitionId(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java:192]: Be careful! The previous method is using the objectDefinitionId1 and the new one is using objectDefinitionId1 or objectDefinitionId2, is that the expected behaviour?

Comment (sergiojimcos) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java:192]: Make sense @ccorreagg!! I suppose that it was using the objectDefintion1, my bad

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 7]
```diff
@@ -1072,19 +1153,38 @@ private void _testPostCustomObjectEntryWithNestedSystemObjectEntry(
...
+		JSONObject systemObjectEntryJSONObject = HTTPTestUtil.invoke(
+			null,
+			StringBundler.concat(
+				jaxRsApplicationDescriptor.getRESTContextPath(),
+				StringPool.SLASH,
+				_getSystemObjectEntryId(
+					jsonObject.getString("id"), manyToOne, objectRelationship)),
+			Http.Method.GET);
+
+		_assertSystemObjectEntryValue(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:1165]: I think this way of validating the system object is a bit weird as you are:

- Validating the emailAddress field of the user account against the emailAddress returned by the GET endpoint
- Validating the custom object field of the user account against the custom field returned by the GET endpoint

What would happen if the related system object entry was not returned in the POST call?

I think we should verify that.

Apart from that, maybe we should modify the `_assertSystemObjectEntryValue` to check the system object in one invocation, not two

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 8]
```diff
@@ -1044,24 +1120,29 @@ private void _testPostCustomObjectEntryWithInvalidNestedSystemObjectEntries(
...
 		Assert.assertEquals("BAD_REQUEST", jsonObject.get("status"));
 	}
 
 	private void _testPostCustomObjectEntryWithNestedSystemObjectEntry(
-			ObjectRelationship objectRelationship, boolean manyToOne)
+			String[] customObjectField, ObjectRelationship objectRelationship,
+			boolean manyToOne)
 		throws Exception {
 
+		JaxRsApplicationDescriptor jaxRsApplicationDescriptor =
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:1134]: This `JaxRsApplicationDescriptor` is being used 24 lines after, please, sort the code

**Review by ccorreagg — COMMENTED** [review]
Some comments to review

**Comment by ccorreagg** [comment]
Rename _CUSTOM_OBJECT_FIELD_NAME_x

**Comment by sergiojimcos** [comment]
ci:test:sf

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java [hunk 9]
```diff
@@ -856,6 +882,56 @@ private String _getLocation(String name, long primaryKey) {
 			_getLocation(name), StringPool.SLASH, primaryKey);
 	}
 
+	private List<ObjectFieldSetting> _getObjectFieldSettings(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:868]: To simplify, move this method to the Util class, then you don't have to worry about the ObjectFieldSettings list (you or whoever uses the Util class)

Comment (sergiojimcos) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java:868]: Make sense @ccorreagg!!

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
> however I just checked that some arguments are not alphabetically sorted of the new methods, maybe we need to be careful with that if we want to avoid resending requests

I'm going to check that and see if there is something that is not sorted

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward:force

**Pull request closed without merging**